sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -590,7 +591,11 @@ object ScalaReflection extends ScalaReflection {
               "cannot be used as field name\n" + walkedTypePath.mkString("\n"))
           }
 
-          val fieldValue = Invoke(inputObject, fieldName, dataTypeFor(fieldType))
+          // primitive take only non-null or struct takes non-null object guarded by isNull
```

Comment:
The concept of `None` can't well fit with Tuple, except for Tuple1.

For a Tuple 2, for example, we encoder it as a row of 2 columns. If it is None, should we encode it as `[null]` or `[null, null]`? Conceptually, `[null]` looks the correct answer. However, practically it becomes a row with only one column and there is a conflict in data format.

Currently, if given a None for a Tuple2 data, we will encode it as `[null, null]`, as you seen in the test @ueshin mentioned.